src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,14 +6,34 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonl = new int[matrix.length];
+        for (int i = 0; i < matrix.length && matrix[i].length > i; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < matrix.length; i++) {
```
we use square matrix in this task so we shouldn't use this `&& matrix[i].length > i`

Reply:
Rectangular matrix is ​​not necessarily square)